managed-ledger/src/main/java/org/apache/bookkeeper/mledger/impl/ManagedLedgerFactoryImpl.java

```diff
@@ -443,7 +443,8 @@ public void asyncOpenReadOnlyManagedLedger(String managedLedgerName,
 
         }).exceptionally(e -> {
             log.error("[{}] Failed to initialize Read-only managed ledger", managedLedgerName, e);
-            callback.openReadOnlyManagedLedgerFailed((ManagedLedgerException) e.getCause(), ctx);
+            callback.openReadOnlyManagedLedgerFailed(
+                    new ManagedLedgerException.ManagedLedgerNotFoundException(e.getMessage()), ctx);
```

Comment:
Why do we convert all of the exceptions to `ManagedLedgerNotFoundException`?

Reply:
It was my oversight, how about changing it to:
```java
callback.openReadOnlyManagedLedgerFailed(
                    new ManagedLedgerException(e.getMessage()), ctx);
```